### PR TITLE
Release Notes draft

### PR DESCRIPTION
**Pull request by nikita-malininn**
### Changes

- Release Notes update

### CODE REVIEW

**Comment by nikita-malininn** [comment]
@vshampor, @MaximProshin

File: ReleaseNotes.md [hunk 1]
```diff
@@ -1,5 +1,55 @@
 # Release Notes
 
+## New in Release 2.5.0
+Compression-aware training:
+- New Features:
+  - Added Patcher class from OTX, which is used for patching the models that can not be wrapped properly.
```
Comment (vshampor) [inline ReleaseNotes.md]: Added `nncf.common.utils.patcher.Patcher` - this class can be used to patch methods on live PyTorch model objects with wrappers such as `nncf.torch.dynamic_graph.context.no_nncf_trace` when doing so in the model code is not possible (e.g. if the model comes from an external library package).

File: ReleaseNotes.md [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+  - (PyTorch, Tensorflow) Added `strip` that convert compressed model to inference in backend format without NNCF specific operations (with int4 support).
```
Comment (vshampor) [inline ReleaseNotes.md]: Remove `(with int4 support)`.

Compression controllers of the `nncf.api.compression.CompressionAlgorithmController` class now have a `.strip()` method that will return the compressed model object with as many custom NNCF additions removed as possible while preserving the functioning of the model object as a compressed model.

File: ReleaseNotes.md [hunk 3]
```diff
@@ -1,5 +1,55 @@
...
+- Fixes:
+  - Fixed traced tensors to implement the YOLOv8 from Ultralytics.
```
Comment (vshampor) [inline ReleaseNotes.md]: This is (PyTorch)

File: ReleaseNotes.md [hunk 4]
```diff
@@ -1,5 +1,55 @@
...
+  - Fixed statistics computation for pruned layers.
+- Improvements:
+  - Extension of attributes (`transpose/permute/getitem`) for pruning node selector.
+  - NNCFNetwork was refactored from a wrapper-approach to a mixin-like approach.
```
Comment (vshampor) [inline ReleaseNotes.md]: (PyTorch)

File: ReleaseNotes.md [hunk 5]
```diff
@@ -1,5 +1,55 @@
...
+  - Added configuration for Accuracy Aware (pruning + quantization) for Resnet18 on CIFAR10.
```
Comment (vshampor) [inline ReleaseNotes.md]: (PyTorch) Added config file for ResNet18 accuracy-aware pruning + quantization on CIFAR10

File: ReleaseNotes.md [hunk 6]
```diff
@@ -1,5 +1,55 @@
...
+  - Added average pool 3d-like ops to pruning maskAdded Conv3d for overflow fix.
+  - Fixed JIT-traceable PyTorch models with internal patching.
```
Comment (vshampor) [inline ReleaseNotes.md]: (PyTorch)

File: ReleaseNotes.md [hunk 7]
```diff
@@ -1,5 +1,55 @@
...
+  - Added Patcher class from OTX, which is used for patching the models that can not be wrapped properly.
+  - (PyTorch, Tensorflow) Added `strip` that convert compressed model to inference in backend format without NNCF specific operations (with int4 support).
+- Fixes:
+  - Fixed traced tensors to implement the YOLOv8 from Ultralytics.
+  - Fixed statistics computation for pruned layers.
+- Improvements:
+  - Extension of attributes (`transpose/permute/getitem`) for pruning node selector.
+  - NNCFNetwork was refactored from a wrapper-approach to a mixin-like approach.
+  - Added configuration for Accuracy Aware (pruning + quantization) for Resnet18 on CIFAR10.
+  - Added average pool 3d-like ops to pruning maskAdded Conv3d for overflow fix.
```
Comment (vshampor) [inline ReleaseNotes.md]: Two lines stuck together. Is this (PyTorch)?

File: ReleaseNotes.md [hunk 8]
```diff
@@ -1,5 +1,55 @@
...
+  - Fixed JIT-traceable PyTorch models with internal patching.
+  - Added `__matmul__` magic functions to the list of patched ops (for SwinTransformer by Microsoft).
```
Comment (vshampor) [inline ReleaseNotes.md]: (PyTorch)

File: ReleaseNotes.md [hunk 9]
```diff
@@ -1,5 +1,55 @@
...
+  - Added `set_log_file()` method was added for NNCF logger.
```
Comment (vshampor) [inline ReleaseNotes.md]: `nncf.set_log_file(...)` can now be used to set location of the NNCF log file.

File: ReleaseNotes.md [hunk 10]
```diff
@@ -1,5 +1,55 @@
...
+  - Added support for pruning of `torch.nn.functional.pad` operation.
+  - Added `torch.baddbmm` as an alias for the matmul metatype for quantization purposes.
```
Comment (vshampor) [inline ReleaseNotes.md]: (PyTorch)

File: ReleaseNotes.md [hunk 11]
```diff
@@ -1,5 +1,55 @@
...
+- Requirements:
+  - Updated ONNX version (1.13)
+- Docs:
+  - Multiple docs updates (fixes, improvements, etc.).
+
+Post-training compression:
+- Features:
+  - Added `"overflow_fix"` parameter support (improved accuracy for optimized model for affected devices).
+  - Added `"model_type"` parameter support that allows reach better accuracy for specific model architectures (e.g. transformers).
+  - Added `"inplace_statistics"` support (reduce memory footprint during optimization).
```
Comment (vshampor) [inline ReleaseNotes.md]: Might want to mention *where* the parameters were added to.

File: ReleaseNotes.md [hunk 12]
```diff
@@ -1,5 +1,55 @@
...
+  - Added `nncf.parameters.IgnoredScope` parameter support that allows exclude layers from optimization scope in different ways.
```
Comment (vshampor) [inline ReleaseNotes.md]: Mention where this new parameter appears in API calls.

File: ReleaseNotes.md [hunk 13]
```diff
@@ -1,5 +1,55 @@
...
+  - (OpenVINO) Added Quantization with accuracy control algorithm.
+  - (OpenVINO) Added YOLOv8 examples for [quantization](examples/post_training_quantization/openvino/yolov8) & [quantization with accuracy control](examples/post_training_quantization/openvino/yolov8_quantize_with_accuracy_control).
+  - (OpenVINO, ONNX) Added BiasCorrection algorithm support.
+  - (OpenVINO) Added support for dynamic-shape models.
+  - (OpenVINO) Added GRU/LSTM quantization support.
+  - (OpenVINO) Added quantizer scales unification.
+  - (PyTorch) Added MinMaxAlgorithm support.
+- Fixes:
+  - Fixed several bugs which lead to performance drop (added a number of patterns, corrected quantizer parameters calculation, etc.).
+  - Checking correct ONNX opset version via the `nncf.quantize(...)`. For now, models with opset < 13 optimizes correctly in per-tensor quantization.
```
Comment (vshampor) [inline ReleaseNotes.md]: (ONNX) `Fixed checking for correct ONNX opset version in `nncf.quantize(...)` calls with ONNX models - now models with opset < 13 will be optimized correctly during per-tensor quantization.

File: ReleaseNotes.md [hunk 14]
```diff
@@ -1,5 +1,55 @@
...
+  - (OpenVINO) Added support of the Depthwise & GroupConvolution layers support for correct quantization.
+  - (OpenVINO) Added Quantization with accuracy control algorithm.
+  - (OpenVINO) Added YOLOv8 examples for [quantization](examples/post_training_quantization/openvino/yolov8) & [quantization with accuracy control](examples/post_training_quantization/openvino/yolov8_quantize_with_accuracy_control).
+  - (OpenVINO, ONNX) Added BiasCorrection algorithm support.
+  - (OpenVINO) Added support for dynamic-shape models.
+  - (OpenVINO) Added GRU/LSTM quantization support.
+  - (OpenVINO) Added quantizer scales unification.
+  - (PyTorch) Added MinMaxAlgorithm support.
+- Fixes:
+  - Fixed several bugs which lead to performance drop (added a number of patterns, corrected quantizer parameters calculation, etc.).
```
Comment (vshampor) [inline ReleaseNotes.md]: Is this (ONNX)?

File: ReleaseNotes.md [hunk 15]
```diff
@@ -1,5 +1,55 @@
...
+  - Checking correct ONNX opset version via the `nncf.quantize(...)`. For now, models with opset < 13 optimizes correctly in per-tensor quantization.
+  - Fixed `ignored_scope` attribute behaviour for weights. Now, the weighted layers excludes from optimization scope correctly.
```
Comment (vshampor) [inline ReleaseNotes.md:21]: Is this (PyTorch), (ONNX), common?

File: ReleaseNotes.md [hunk 16]
```diff
@@ -1,5 +1,55 @@
...
+- Improvements:
+  - Improvements for pipeline speed up (transformations usage in ModelTransformer).
```
Comment (vshampor) [inline ReleaseNotes.md]: Which pipelines were improved? Is this common?

**Review by vshampor — CHANGES_REQUESTED** [review]

File: ReleaseNotes.md (hunk 16 above)
Comment (nikita-malininn) [reply ReleaseNotes.md]: Common

File: ReleaseNotes.md (hunk 15 above)
Comment (nikita-malininn) [reply ReleaseNotes.md:21]: Common

File: ReleaseNotes.md (hunk 1 above)
Comment (nikita-malininn) [reply ReleaseNotes.md]: Is it only for PyTorch, bu the way?

File: ReleaseNotes.md (hunk 7 above)
Comment (nikita-malininn) [reply ReleaseNotes.md]: Both for PyTorch & Tensorflow

File: ReleaseNotes.md (hunk 14 above)
Comment (nikita-malininn) [reply ReleaseNotes.md]: Common

**Comment by MaximProshin** [comment]
I would reorder it and start from Post-training Quantization (not Compression) as it will be the main feature of this release. I would also clearly state that we formally introduce PTQ support for OpenVINO IR as the first statement and then list the particular features like AAQ as sub-bullets. List the algorithms at the beginning.

File: ReleaseNotes.md [hunk 17]
```diff
@@ -1,5 +1,56 @@
...
+  - (OpenVINO) Added support of the Depthwise & GroupConvolution layers support for correct quantization.
+  - (OpenVINO) Added Quantization with accuracy control algorithm.
+  - (OpenVINO) Added YOLOv8 examples for [quantization](examples/post_training_quantization/openvino/yolov8) & [quantization with accuracy control](examples/post_training_quantization/openvino/yolov8_quantize_with_accuracy_control).
+  - (OpenVINO, ONNX) Added BiasCorrection algorithm support.
+  - (OpenVINO) Added support for dynamic-shape models.
+  - (OpenVINO) Added GRU/LSTM quantization support.
+  - (OpenVINO) Added quantizer scales unification.
+  - (PyTorch) Added MinMaxAlgorithm support.
+- Fixes:
+  - Fixed several bugs which lead to performance drop (added a number of patterns, corrected quantizer parameters calculation, etc.).
```
Comment (MaximProshin) [inline ReleaseNotes.md]: If it's internal bugs, perhaps we could remove it as it's vague.

File: ReleaseNotes.md [hunk 18]
```diff
@@ -1,5 +1,56 @@
...
+  - Added average pool 3d-like ops to pruning mask.
+  - Added Conv3d for overflow fix.
+  - `nncf.set_log_file(...)` can now be used to set location of the NNCF log file.
+  - Added support for pruning of `torch.nn.functional.pad` operation.
+  - (PyTorch) Added `torch.baddbmm` as an alias for the matmul metatype for quantization purposes.
+  - (PyTorch) Added config file for ResNet18 accuracy-aware pruning + quantization on CIFAR10.
+  - (PyTorch) Fixed JIT-traceable PyTorch models with internal patching.
+  - (PyTorch) Added `__matmul__` magic functions to the list of patched ops (for SwinTransformer by Microsoft).
+- Requirements:
+  - Updated ONNX version (1.13)
```
Comment (MaximProshin) [inline ReleaseNotes.md:49]: Add also TF 2.11 (PR should be merged soon)

File: ReleaseNotes.md [hunk 19]
```diff
@@ -1,5 +1,56 @@
...
+- Docs:
+  - Multiple docs updates (fixes, improvements, etc.).
```
Comment (MaximProshin) [inline ReleaseNotes.md]: It looks vague and the documentation update is a normal change. Propose to remove it

File: ReleaseNotes.md [hunk 20]
```diff
@@ -1,5 +1,56 @@
...
+  - (OpenVINO) Added support for dynamic-shape models.
+  - (OpenVINO) Added GRU/LSTM quantization support.
+  - (OpenVINO) Added quantizer scales unification.
+  - (PyTorch) Added MinMaxAlgorithm support.
+- Fixes:
+  - Fixed several bugs which lead to performance drop (added a number of patterns, corrected quantizer parameters calculation, etc.).
+  - Fixed `ignored_scope` attribute behaviour for weights. Now, the weighted layers excludes from optimization scope correctly.
+  - (ONNX) Checking correct ONNX opset version via the `nncf.quantize(...)`. For now, models with opset < 13 optimizes correctly in per-tensor quantization.
+- Improvements:
+  - Improvements for pipeline speed up (transformations usage in ModelTransformer).
```
Comment (MaximProshin) [inline ReleaseNotes.md]: I don't understand it and I wonder if users will understand it. Either remove or try to rephrase.

File: ReleaseNotes.md [hunk 21]
```diff
@@ -1,5 +1,56 @@
...
+  - Added improvements for statistic collection process (collect statistics only once).
+  - (PyTorch, OpenVINO, ONNX) Introduced unified quantizer parameters calculation.
+  - (OpenVINO) Added support for per-tensor & per-channel quantization.
+  - (OpenVINO) Aligned parameters for FP16 models.
+  - (OpenVINO) Aligned patterns between POT & NNCF.
```
Comment (MaximProshin) [inline ReleaseNotes.md]: I would remove it or rephrase it like "Added quantization patterns for ..."

File: ReleaseNotes.md [hunk 22]
```diff
@@ -1,5 +1,56 @@
 # Release Notes
 
+## New in Release 2.5.0
```
Comment (MaximProshin) [inline ReleaseNotes.md:3]: Not covered: 
- PTQ support for PyTorch: added MinMax and FBC algorithms
- prepare_for_inference API 
- JPQD support for BERT, Wave2VecV2, Swin, ViT, DistilBERT, CLIP, MobileBERT

Comment (MaximProshin) [reply ReleaseNotes.md:3]: resolved

File: ReleaseNotes.md [hunk 23]
```diff
@@ -1,5 +1,55 @@
...
+Post-training Quantization:
+- Features:
+  - Added `"overflow_fix"` parameter support (improved accuracy for optimized model for affected devices).
+  - Added `"model_type"` parameter support that allows reach better accuracy for specific model architectures (e.g. transformers).
+  - Added `"inplace_statistics"` support (reduce memory footprint during optimization).
+  - Added `nncf.parameters.IgnoredScope` parameter support that allows exclude layers from optimization scope in different ways.
+  - (OpenVINO) Introduced post-training support for backend.
```
Comment (MaximProshin) [inline ReleaseNotes.md]: I would put it as the first bullet after "Features" and rephrase it to 
- Added Post-training Quantization support for OpenVINO IR (OpenVINO backend).

File: ReleaseNotes.md (hunk 18 above)
Comment (MaximProshin) [reply ReleaseNotes.md:49]: resolved

File: ReleaseNotes.md (hunk 19 above)
Comment (MaximProshin) [reply ReleaseNotes.md]: resolved

File: ReleaseNotes.md (hunk 17 above)
Comment (MaximProshin) [reply ReleaseNotes.md]: resolved

File: ReleaseNotes.md (hunk 20 above)
Comment (MaximProshin) [reply ReleaseNotes.md]: resolved

File: ReleaseNotes.md (hunk 21 above)
Comment (MaximProshin) [reply ReleaseNotes.md]: resolved

File: ReleaseNotes.md (hunk 1 above)
Comment (vshampor) [reply ReleaseNotes.md]: > Is it only for PyTorch, bu the way?

Not only for PyTorch, but it only currently has uses for PyTorch.

File: ReleaseNotes.md (hunk 22 above)
Comment (nikita-malininn) [reply ReleaseNotes.md:3]: > * PTQ support for PyTorch: added MinMax and FBC algorithms

Added, only MinMax algo
> * prepare_for_inference API

It's added as `strip()`.

> * JPQD..

Added.

File: ReleaseNotes.md (hunk 23 above)
Comment (nikita-malininn) [reply ReleaseNotes.md]: Done

File: ReleaseNotes.md (hunk 12 above)
Comment (nikita-malininn) [reply ReleaseNotes.md]: Done
